Module_4/FileSystemVisitorLibrary/FileSystemVisitor.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+
+namespace FileSystemVisitorLibrary
+{
+    public class FileSystemVisitor
+    {
+        private readonly Func<FileSystemInfo, bool> _filterFunc;
```

Comment:
It's clear that this is a `Func` when we look at the type of the variable.

```suggestion
        private readonly Func<FileSystemInfo, bool> _filter;
```